src/main/java/hivemall/mix/server/MixServerHandler.java

```diff
@@ -57,7 +61,23 @@ protected void channelRead0(ChannelHandlerContext ctx, MixMessage msg) throws Ex
             case argminKLD: {
                 SessionObject session = getSession(msg);
                 PartialResult partial = getPartialResult(msg, session);
-                mix(ctx, msg, partial, session);
+                if (mix(msg, partial)) {
+                    try {
+                        partial.lock();
```

Comment:
ISTM that, since mix() updates `PartialResult`s, the three PartialResult getters have inconsistent views in case of no sync.